### ISSUE TITLE
Unable to access internet from Windows Container but can ping internal IP addresses

### ISSUE THREAD

**Issue by JackUkleja** [docker/for-win#1397]
### Expected behavior
To be able to fully access both internal and external networks (internet) from a Windows container, with default installation/settings.

### Actual behavior
Container is only able to ping internal IP addresses (external ping are blocked on our corporate network so I cant test that). Unable to access internal IP addresses using other protocols (e.g. DNS or HTTP) or access internet at all.

### Information

  - Diagnostic ID:  5E6852F7-C960-4284-BB4B-05D29BF181CA/2017-12-08_12-44-09
  - Host: Windows 10 Pro, 1703 (Build 15063.726) 

```
PS C:\Users\uklejja> docker run microsoft/windowsservercore ipconfig /all

Windows IP Configuration

   Host Name . . . . . . . . . . . . : 758fbc9d2cb5
   Primary Dns Suffix  . . . . . . . :
   Node Type . . . . . . . . . . . . : Hybrid
   IP Routing Enabled. . . . . . . . : No
   WINS Proxy Enabled. . . . . . . . : No
   DNS Suffix Search List. . . . . . : corp.local

Ethernet adapter Ethernet 2:

   Connection-specific DNS Suffix  . : corp.local
   Description . . . . . . . . . . . : Microsoft Hyper-V Network Adapter #2
   Physical Address. . . . . . . . . : 00-15-5D-7D-B5-C6
   DHCP Enabled. . . . . . . . . . . : Yes
   Autoconfiguration Enabled . . . . : Yes
   Link-local IPv6 Address . . . . . : fe80::b5d1:b983:2224:2fc8%5(Preferred)
   IPv4 Address. . . . . . . . . . . : 172.30.95.100(Preferred)
   Subnet Mask . . . . . . . . . . . : 255.255.240.0
   Default Gateway . . . . . . . . . : 172.30.80.1
   DHCPv6 IAID . . . . . . . . . . . : 83891549
   DHCPv6 Client DUID. . . . . . . . : 00-01-00-01-21-BB-A9-17-00-15-5D-7D-B5-C6
   DNS Servers . . . . . . . . . . . : 172.30.80.1
                                       192.168.18.210
                                       192.168.18.201
   NetBIOS over Tcpip. . . . . . . . : Disabled
```

On host OS:
```
ipconfig /all

Ethernet adapter vEthernet (HNS Internal NIC):

   Connection-specific DNS Suffix  . :
   Description . . . . . . . . . . . : Hyper-V Virtual Ethernet Adapter #2
   Physical Address. . . . . . . . . : 00-15-5D-7D-B4-33
   DHCP Enabled. . . . . . . . . . . : Yes
   Autoconfiguration Enabled . . . . : Yes
   Link-local IPv6 Address . . . . . : fe80::3d55:e332:1afa:2b8d%24(Preferred)
   IPv4 Address. . . . . . . . . . . : 172.30.80.1(Preferred)
   Subnet Mask . . . . . . . . . . . : 255.255.240.0
   Default Gateway . . . . . . . . . :
   DHCPv6 IAID . . . . . . . . . . . : 402658653
   DHCPv6 Client DUID. . . . . . . . : 00-01-00-01-21-7D-90-3D-98-90-96-D7-FD-97
   DNS Servers . . . . . . . . . . . : fec0:0:0:ffff::1%1
                                       fec0:0:0:ffff::2%1
                                       fec0:0:0:ffff::3%1
   NetBIOS over Tcpip. . . . . . . . : Enabled

Ethernet adapter Ethernet:

   Connection-specific DNS Suffix  . : corp.local
   Description . . . . . . . . . . . : Intel(R) Ethernet Connection I217-LM
   Physical Address. . . . . . . . . : 98-90-96-D7-FD-97
   DHCP Enabled. . . . . . . . . . . : Yes
   Autoconfiguration Enabled . . . . : Yes
   Link-local IPv6 Address . . . . . : fe80::4dc:522a:63be:e582%15(Preferred)
   IPv4 Address. . . . . . . . . . . : 192.168.18.117(Preferred)
   Subnet Mask . . . . . . . . . . . : 255.255.252.0
   Lease Obtained. . . . . . . . . . : Friday, 8 December 2017 12:40:28 PM
   Lease Expires . . . . . . . . . . : Saturday, 16 December 2017 12:40:28 PM
   Default Gateway . . . . . . . . . : 192.168.18.11
   DHCP Server . . . . . . . . . . . : 192.168.18.210
   DHCPv6 IAID . . . . . . . . . . . : 110661782
   DHCPv6 Client DUID. . . . . . . . : 00-01-00-01-21-7D-90-3D-98-90-96-D7-FD-97
   DNS Servers . . . . . . . . . . . : 192.168.18.210
                                       192.168.18.201
   NetBIOS over Tcpip. . . . . . . . : Enabled
```

### Steps to reproduce the behavior

```
docker run microsoft/windowsservercore ping 192.168.18.210
Pinging 192.168.18.210 with 32 bytes of data:
Reply from 192.168.18.210: bytes=32 time=1ms TTL=127
Reply from 192.168.18.210: bytes=32 time=2ms TTL=127

docker run microsoft/windowsservercore ping google.com
Ping request could not find host google.com. Please check the name and try again.

docker run microsoft/windowsservercore powershell invoke-webrequest -uri google.com
invoke-webrequest : The remote name could not be resolved: 'google.com'
```

I have run wireshark and I can see the pings going out on the wire via NAT, but both DNS and HTTP requests never even appear on the wire.

**Comment by LordMike** [comment]
You could change the network on the container to use a bridged network (unless a private network is explicitly required in your setup). In the Hyper-V console, under virtual switches, you should be able to find a network named "nat". It will be set to Private or Internal Networking. Setting it to External Networking makes i bridged. 

To use private/internal networking modes, you need to share your network connection from the host.

**Comment by JackUkleja** [comment]
@LordMike "Share your network connection from the host" <- How do I do this? Is this with ICS?

**Comment by JackUkleja** [comment]
I have got several developers at the same company, all suffering from this issue. We have tried Windows 10, 1703 and 1709.

Conversely we have a developer whose installation of Docker is working fine with the default nat network.

Unfortunately I cannot locate the difference between the two setups. This is driving me crazy because we are basically unable to roll out Docker into the development pipeline because we can't get the networking to work reliably on all PCs!! :-(

It's possible the corporate SOE is interfering with the networking (virus scanners etc), but then how is it possible that some of the PCs _are_ working with NAT?

Basically the default nat (on the malfunctioning Docker installs) only lets me reach IP addresses on the container's subnet, and the host NIC subnet, but nowhere else.

**Comment by JackUkleja** [comment]
Long shot but could be related to this: https://github.com/docker/for-win/issues/1166

On the 'working' PC I _do_ see this 'ics' network below which I don't see on the malfunctioning PCs:
```
 NAME                DRIVER              SCOPE
 Default Switch      ics                 local
```
Also I note on my corporate PCs, ICS is a disabled by group policy. Could this be a problem?

**Comment by LordMike** [comment]
Things I'd try:

* Reset the Docker installation. They have an option to do this in the tray. Tray -> Settings -> Reset -> Reset to factory -- it's possible some switch setup failed from the beginning. This option will delete the VM (any data in it too), the switches and set it all up again.

* My previous suggestion reg. changing the switch type is done through the Hyper-V management console -- but now I realize this might be horrible advice, so don't do this. I can no longer reach my docker installation this way :P (a Reset like above fixes it)

#### Verification.

Note: My network driver in docker is `bridge`, might want to look into that. Below are some outputs of various configurations.

When I check my routes (host, windows 10), I get the following. I'd verify the LAN connections route is correct, and that any traffic to your DockerNAT is routed correctly. I do have another switch, but it shouldn't affect this setup.

```
IPv4 Route Table
===========================================================================
Active Routes:
Network Destination        Netmask          Gateway       Interface  Metric
          0.0.0.0          0.0.0.0       10.64.11.1     10.64.11.138     25 <-- Default route
        10.0.75.0    255.255.255.0         On-link         10.0.75.1    271 <-- DockerNAT
        10.0.75.1  255.255.255.255         On-link         10.0.75.1    271 <-- DockerNAT
      10.0.75.255  255.255.255.255         On-link         10.0.75.1    271 <-- DockerNAT
       10.64.11.0    255.255.255.0         On-link      10.64.11.138    281 <-- LAN, Ethernet
     10.64.11.138  255.255.255.255         On-link      10.64.11.138    281 <-- LAN, Ethernet
     10.64.11.255  255.255.255.255         On-link      10.64.11.138    281 <-- LAN, Ethernet
        127.0.0.0        255.0.0.0         On-link         127.0.0.1    331 <-- Loopback
        127.0.0.1  255.255.255.255         On-link         127.0.0.1    331 <-- Loopback
  127.255.255.255  255.255.255.255         On-link         127.0.0.1    331 <-- Loopback
    172.23.175.96  255.255.255.240         On-link     172.23.175.97    271 <-- "Default Switch"
    172.23.175.97  255.255.255.255         On-link     172.23.175.97    271 <-- "Default Switch"
   172.23.175.111  255.255.255.255         On-link     172.23.175.97    271 <-- "Default Switch"
        224.0.0.0        240.0.0.0         On-link         127.0.0.1    331 <-- Loopback
        224.0.0.0        240.0.0.0         On-link      10.64.11.138    281 <-- LAN, Ethernet
        224.0.0.0        240.0.0.0         On-link     172.23.175.97    271 <-- "Default Switch"
        224.0.0.0        240.0.0.0         On-link         10.0.75.1    271 <-- DockerNAT
  255.255.255.255  255.255.255.255         On-link         127.0.0.1    331 <-- Loopback
  255.255.255.255  255.255.255.255         On-link      10.64.11.138    281 <-- LAN, Ethernet
  255.255.255.255  255.255.255.255         On-link     172.23.175.97    271 <-- "Default Switch"
  255.255.255.255  255.255.255.255         On-link         10.0.75.1    271 <-- DockerNAT
```

Inside docker, I get this:
```
PS C:\<--snip-->> docker run -it --rm bash
bash-4.4# ifconfig
eth0      Link encap:Ethernet  HWaddr 02:42:AC:11:00:02
          inet addr:172.17.0.2  Bcast:0.0.0.0  Mask:255.255.0.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:9 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:758 (758.0 B)  TX bytes:0 (0.0 B)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

bash-4.4# ping google.dk
PING google.dk (216.58.201.163): 56 data bytes
64 bytes from 216.58.201.163: seq=0 ttl=37 time=12.388 ms
64 bytes from 216.58.201.163: seq=1 ttl=37 time=13.584 ms
64 bytes from 216.58.201.163: seq=2 ttl=37 time=13.507 ms
^C
--- google.dk ping statistics ---
3 packets transmitted, 3 packets received, 0% packet loss
round-trip min/avg/max = 12.388/13.159/13.584 ms
bash-4.4#
```

I see the above in Wireshark on the host, coming to and from my LAN IP:
![image](https://user-images.githubusercontent.com/1027111/34259459-9a88b158-e662-11e7-8bc9-2a6ef16dbaae.png)

From `docker network ls`:

```
PS C:\<--snip-->> docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
cb081d64b986        bridge              bridge              local
cce1bdea04fb        host                host                local
c4d3fe20ed9a        none                null                local
```

From `docker network inspect bridge`:
```
> docker network inspect bridge
[
    {
        "Name": "bridge",
        "Id": "cb081d64b986ded5438a5ce346dcc7a855be490af002ce4ba375983783afdd8d",
        "Created": "2017-12-21T14:10:18.3694205Z",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.17.0.0/16"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "654bd68c91b1396601e066ba6fae482d3aac3b364bfbd1790cfcc9a58f32c5ed": {
                "Name": "keen_golick",
                "EndpointID": "4e7e6b4231ccd5e8594091f479320bbdbe90501b29b3e0d47d30c8e71a6e3352",
                "MacAddress": "02:42:ac:11:00:02",
                "IPv4Address": "172.17.0.2/16",
                "IPv6Address": ""
            }
        },
        "Options": {
            "com.docker.network.bridge.default_bridge": "true",
            "com.docker.network.bridge.enable_icc": "true",
            "com.docker.network.bridge.enable_ip_masquerade": "true",
            "com.docker.network.bridge.host_binding_ipv4": "0.0.0.0",
            "com.docker.network.bridge.name": "docker0",
            "com.docker.network.driver.mtu": "1500"
        },
        "Labels": {}
    }
]
```

**Comment by LordMike** [comment]
Reg. ICS and in general, if you have a PC at your office that is not domain joined and therefore under the company policy, it could be useful to test if it works out of the box. If it doesn't, it could indicate a policy issue.

**Comment by JackUkleja** [comment]
I found the problem. All my problems were caused by Symantec Endpoint Protection 14 we have installed on the corporate PCs. It interferes with the container networking (although did not 100% block all traffic). In spite of [claims SEP works OK with Docker](https://support.symantec.com/en_US/article.TECH236867.html), my experience is that it doesn't. Even after updating to latest builds.

After completely disabling Symantec everything worked perfectly.
